Removes Underscore dependency

grant opened the pull request:
Goal:
- Remove 1548 lines of Apps Script to improve library parse + load time.
- Remove copypasted `underscore@1.8.3` library from scm.

This PR:
- Removes `_` with `gulpfile`/`package.json` references
- Use native util methods over `_`
  - Remove `isEmpty_`
- Regens docs

Feel free to comment or modify.

erickoledadevrel commented on src/Utilities.gs:
```diff
@@ -39,24 +39,12 @@ function buildUrl_(url, params) {
   Object.keys(params).forEach(function(name) {
     var value = params[name];
-    if (isEmpty_(value)) {
+    if (!value) {
```
Not functionally equivalent to isEmpty_(), but I don't think any of my validations rely on detecting empty arrays or objects.

grant replied:
Good point.

erickoledadevrel commented:
Thanks for the PR! Service.gs still uses _.extend() in a number of places however, and you'll need to find a replacement for that.

grant commented:
PTAL.

- Added `extend_()`
  - Note: Only accepts 2 objects
  - Added Utility tests for this method

```js
npm run test

> mocha
  Service
    #getToken()
      ✓ should load from the in-memory store
      ✓ should load from the cache and set the in-memory store
      ✓ should load from the properties and set the cache and in-memory store
    #saveToken_()
      ✓ should save the token to the properties, cache, and in-memory store
    #reset()
      ✓ should delete the token from properties, cache, and in-memory store

  Utilities
    #extend_()
      ✓ should extend (left) an object
      ✓ should extend (right) an object
      ✓ should extend (merge) an object


  8 passing (10ms)
```

erickoledadevrel commented:
Looks great, thanks Grant!
